src/go/rpk/pkg/cli/topic/describe.go

```diff
@@ -53,107 +66,116 @@ partitions section. By default, the summary and configs sections are printed.
 			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
 			defer cl.Close()
 
-			topic := topicArg[0]
+			adm, err := kafka.NewAdmin(fs, p)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer adm.Close()
+
+			if re {
+				topicArg, err = regexTopics(adm, topicArg)
+				out.MaybeDie(err, "unable to filter topics by regex: %v", err)
+			}
 
 			// By default, if neither are specified, we opt in to
 			// the config section only.
 			if !summary && !configs && !partitions {
 				summary, configs = true, true
 			}
-			if all {
```

Comment:
We should add back this option. Some tests fail because the flag `--all` stopped working.

Reply:
Great catch. Fixed.